### PR TITLE
[doc] Cleanup and fixes for install guide.

### PR DESCRIPTION
**Pull request by jadephilipoom**
Adjust the install guide pages based on feedback from a live-install at CHES and on a read-through:
- Reverses the order of the Verilator install and the software build, since Bazel tests essentially require Verilator
- Changes the tl;dr section of the software build page to something that should succeed on the first try (the previous example required Verible)
- Adds troubleshooting notes and information about RAM requirements
- Adds notes and time estimates for long-running steps
- Updates information about the CI setup (we now use Ubuntu 20.04, not 18.04)
- Removes `$` before console commands to make them easier to copy-paste

### CODE REVIEW

File: doc/getting_started/_index.md [hunk 1]
```diff
@@ -50,78 +46,88 @@ You can then **skip to step 4** (building software).
...
+Finally, update the symbolic links for `gcc` and `g++` using these commands:
 
 ```console
-$ cd $REPO_TOP
-$ pip3 install --user -r python-requirements.txt
+sudo ln -sf /usr/bin/gcc-9 /usr/bin/gcc
+sudo ln -sf /usr/bin/g++-9 /usr/bin/g++
 ```
 
 ## Step 3: Install the LowRISC RISC-V Toolchain
```
Comment (drewmacrae) [inline doc/getting_started/_index.md:120]: I think you can skip this step as Bazel will manage it all under the hood now, but I'm not confident that no one uses the toolchain outside of Bazel, or doesn't benefit from a locally installed copy.

Comment (jadephilipoom) [reply doc/getting_started/_index.md:120]: Are you referring to the toolchain installation or the GCC linking above, or both?

File: doc/getting_started/build_sw.md [hunk 2]
```diff
@@ -13,12 +13,18 @@ All OpenTitan software is built with [Bazel](https://bazel.build/).
 Additionally, _most_ tests may be run with Bazel too.
 
 ## TL;DR
-To install the correct version of bazel, build, and run all on-host tests you can simply run:
+To install the correct version of bazel, build, and run a single test with Verilator, run:
 
 ```console
-$REPO_TOP/bazelisk.sh test //... --test_tag_filters=-cw310,-verilator --disk_cache=~/bazel_cache
+$REPO_TOP/bazelisk.sh test --test_output=streamed --test_tag_filters=verilator --disk_cache=~/bazel_cache //sw/device/tests:uart_smoketest
```
Comment (drewmacrae) [inline doc/getting_started/build_sw.md:19]: NIT: This invocation is correct but has a trap in it:
The macro `//sw/device/tests:uart_smoketest` will build all its targets including software for the fpga and dv. (No big deal but it takes longer on small non-parallel systems like Jake.) If someone tries to use a similar invocation to build SW for other targets, they might be surprised to see verilator running to support the verilator tests in this macro (which then won't be executed because of the `--test_tag_filter`).

There's a whole set of traps related to how builds and tests are filtered: `--build_tag_filters` and `--build_tests_only` prune targets matched by wildcards, not dependencies so they wont have an effect here and are useful but not as relevant as they might seem and don't offer an answer.

Discovering the platform dependent test's labels is more difficult. They don't appear in our source code and tab completion doesn't work for them, but they're the best way to run single tests and have the build behave as you'd expect:
`$REPO_TOP/bazelisk.sh test --test_output=streamed --disk_cache=~/bazel_cache //sw/device/tests:uart_smoketest_sim_verilator`
This will also help people know those targets exist and how to use them.

File: doc/getting_started/_index.md (hunk 1 above)
Comment (drewmacrae) [reply doc/getting_started/_index.md:120]: Just the toolchain installation. Our unit tests use the host's GCC and we have features that depend on GCC-9 plus to compile without errors.

**Review by drewmacrae — APPROVED** [review]

File: doc/getting_started/_index.md [hunk 3]
```diff
@@ -50,78 +46,88 @@ You can then **skip to step 4** (building software).
 If you do have Linux, you are still welcome to try the Docker container.
 However, as the container option is currently experimental, we recommend following the steps below to build manually if you plan on being a long-term user or contributor for the project.
 
-Our continuous integration setup runs on Ubuntu 18.04 LTS, which gives us the most confidence that this distribution works out of the box.
+Our continuous integration setup runs on Ubuntu 20.04 LTS, which gives us the most confidence that this distribution works out of the box.
 We do our best to support other distributions, but cannot guarantee they can be used "out of the box" and might require updates of packages.
 Please file a [GitHub issue](https://github.com/lowRISC/opentitan/issues) if you need help or would like to propose a change to increase compatibility with other distributions.
 
+You will need at least 7GB of RAM in order to build the Verilator simulation.
+If you have an FPGA, this constraint does not apply.
```
Comment (timothytrippel) [inline doc/getting_started/_index.md:54]: Unless you are building the bitstream, then there is probably a memory constraint (from recent experiments we know 4GB is not enough,  though @dmcardle will probably have better numbers here). If you are just downloading a bitstream from our cloud bucket then obviously the mem limit doesn't apply.

File: doc/getting_started/_index.md (hunk 1 above)
Comment (timothytrippel) [reply doc/getting_started/_index.md:120]: I think its convenient to have a locally installed toolchain for debugging purposes, even though bazel should pick up the right version. Another option is to use [`update-alternatives`](http://charette.no-ip.com:81/programming/2011-12-24_GCCv47/) to manage the versions instead of forced symlinking, but I do like the simplicity of what you currently have, so I would probably just keep this approach.

**Review by timothytrippel — APPROVED** [review]

File: doc/getting_started/_index.md (hunk 1 above)
Comment (jadephilipoom) [reply doc/getting_started/_index.md:120]: Alright, based on this discussion I think I'll leave the local installation instructions for now.

File: doc/getting_started/build_sw.md (hunk 2 above)
Comment (jadephilipoom) [reply doc/getting_started/build_sw.md:19]: Okay, sounds good. I'll change the example here to build `uart_smoketest_sim_verilator` and add a little explanation/link to later in the page.

File: doc/getting_started/_index.md (hunk 3 above)
Comment (jadephilipoom) [reply doc/getting_started/_index.md:54]: I'll rephrase this. 7GB is what our CI has, so it should be enough for both.

Comment (moidx) [reply doc/getting_started/_index.md]: nit:  Is it ok to say "at least 7GB of *available* RAM"?

**Review by moidx — APPROVED** [review]

File: doc/getting_started/_index.md (hunk 3 above)
Comment (dmcardle) [reply doc/getting_started/_index.md]: I'm not sure about the memory requirement for Verilator builds, actually. The thing I learned recently is that we need RAM >16 GB to build CW310 bitstreams with Vivado.

**Review by dmcardle — APPROVED** [review]

File: doc/getting_started/_index.md (hunk 3 above)
Comment (jadephilipoom) [reply doc/getting_started/_index.md]: Ouch. I got the 7GB from the fact that our CI runners have 7GB, and the person I was helping install who ran out of RAM said they kept running out until they increased their VM's allowance from 6GB to 8GB. So > 6GB, <= 7GB are the bounds we have for Verilator.

The updated comment here doesn't speculate about how much memory is needed for building the bitstream, since the default setup is not to build it locally and I don't want to put too much detail about custom setups in the "getting started" page.
